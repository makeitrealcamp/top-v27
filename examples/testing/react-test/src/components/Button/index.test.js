import { Button } from '.';
import { cleanup, render, screen } from '@testing-library/react'

describe('Button', () => {

  beforeEach(() => {
    cleanup()
  })

  it('should render correctly', () => {
    render(<Button type='button'>Crear</Button>)
    const buttonElement = screen.getByText('Crear')

    expect(buttonElement.type).toMatch(/button/i)
    expect(buttonElement).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<Button type='button'>Crear</Button>)

    expect(container).toMatchSnapshot()
  })
})