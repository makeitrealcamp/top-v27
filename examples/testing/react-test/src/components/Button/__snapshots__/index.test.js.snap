// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<div>
  <button
    data-testid="button"
    type="button"
  >
    Crear
  </button>
</div>
`;
